### ISSUE TITLE
Querying domains with comments may return blank lines

### ISSUE THREAD

**Issue by ghost** [pi-hole/pi-hole#2856]
**In raising this issue, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`5`

---
**Expected behaviour:**

Querying the block lists (non-exact/partial) should result in the queried domain appearing in the results if it exists, even if it has a comment at the end -- and even if said comment is preceded by more than one space character.

**Actual behaviour:**

A blank line appears instead.

**Steps to reproduce:**

Using [StevenBlack/hosts](https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts), one of the lines contains the following (although this applies to more than that entry):
```
0.0.0.0 s.youtube.com  # Blocking this will interfere with video watching history
```
When querying `s.youtube.com`, either through the GUI or CLI (`pihole query s.youtube.com`), the result is displayed as a blank line. Querying e.g. `ads.youtube.com`, which does not have a comment at the end, works as intended.

**Troubleshooting undertaken, and/or other relevant information:**

I have narrowed the issue down to [line 181 in query.sh](https://github.com/pi-hole/pi-hole/blob/master/advanced/Scripts/query.sh#L181), for me stored at `/opt/pihole/query.sh`. The issue seems to be caused by the occurrence of more than one space character between the domain and `#`.

Replacing `-e "s/:.*[ \\t]/:/g" \` with `-e "s/:\S\+\s\+/:/g" \` in the above mentioned file seems to result in expected behavior.

**Comment by Mcat12** [comment]
That change looks good to me. One note however is that you shouldn't have to escape the plus signs. I would also recommend switching out the one before it (`[ \\t]#.*`) with `\\s*#.*`.

Could you make a PR for this? If not, we can make the change.

**Comment by mmotti** [comment]
I would recommend `[^\S\n]+#.*$`

Matches where whitespace character but not a new line, followed by comments.

**Comment by u5mo5fbh8wh7** [comment]
@Mcat12 I don't get correct output without escaping the plus signs. I could attempt at PR, but I am not overly familiar with git.

@mmotti That is the same line @Mcat12 recommends changing, right? It doesn't seem to work right for me, though. Without escaping the `+`, comments are still present. With escaping, the whole line disappears.

Switching out the two lines in question with the following seems to work for me:
```
s/\s\+#.*$//g
s/:\S\+\s\+/:/g
```
Actually only switching out the first one seems to work, but I don't see why both shouldn't be updated.

**Comment by mmotti** [comment]
You shouldn't be escaping the `+`

I'm not at my laptop to check atm but what is the outcome of using my suggestion with `sed - r`?

**Comment by u5mo5fbh8wh7** [comment]
@mmotti I hear what you're saying, but _not_ escaping didn't work. This was caused by the missing `-r` it seems.

I can't get your proposal to work even with `-r`, though. It seems to eat the entire line.

**Comment by Mcat12** [comment]
Try using single quotes instead of double quotes. Then you don't need to escape as much.

**Comment by mmotti** [comment]
If you give me 30 mins I'll take a look. I know the regex works because I use it with my host generators to remove in line comments

**Comment by u5mo5fbh8wh7** [comment]
@Mcat12 That might also be it.

Please feel free to make the changes yourselves and implement it however you want.

**Comment by mmotti** [comment]
OK - I was wrong - The regexp I suggested is in-fact not compatible with sed. I must be using a different flavour of regex on a day-to-day basis.

If you only want to remove the in-line comments:
`sed -r 's/\s+#.*$//; /^\s*$/d'`

If you want to remove normal comments and in-line comments:
sed 's/\s*#.*$//; /^\s*$/d'

**Comment by omicron-b** [comment]
I think this can be closed as fixed in 5.0 release, works as expected for me
